src/link/Wasm.zig

```diff
@@ -4715,7 +4715,10 @@ fn linkWithLLD(wasm: *Wasm, comp: *Compilation, prog_node: *std.Progress.Node) !
             // report a nice error here with the file path if it fails instead of
             // just returning the error code.
             // chmod does not interact with umask, so we use a conservative -rwxr--r-- here.
-            try std.os.fchmodat(fs.cwd().fd, full_out_path, 0o744, 0);
+            std.os.fchmodat(fs.cwd().fd, full_out_path, 0o744, 0) catch |err| switch (err) {
+                error.OperationNotSupported => unreachable, // Not a symlink.
```

Comment:
OperationNotSupported can also happen if `/proc` is not mounted and the fallback `fchmodat` is used, right?

Not clear to me why you'd want this error to map to unreachable here.  Seems safer to just return the error as the old code did.   (Really, though it seems like failing to chmod the generated Wasm executable could just be ignored, but that doesn't seem like something this PR should be changing....)